src/uri/authority.rs

```diff
@@ -196,28 +203,28 @@ impl Authority {
     ///
     /// # Examples
     ///
-    /// Authority with port
+    /// Authority with port path
     ///
     /// ```
-    /// # use http::uri::Authority;
+    /// # use http::uri::{Authority, Port};
     /// let authority: Authority = "example.org:80".parse().unwrap();
     ///
-    /// assert_eq!(authority.port(), Some(80));
+    /// assert_eq!(authority.port_path(), Port::from_str("80").ok());
     /// ```
     ///
     /// Authority without port
     ///
     /// ```
-    /// # use http::uri::Authority;
+    /// # use http::uri::{Authority, Port};
     /// let authority: Authority = "example.org".parse().unwrap();
     ///
-    /// assert!(authority.port().is_none());
+    /// assert!(authority.port_path().is_none());
     /// ```
-    pub fn port(&self) -> Option<u16> {
-        let s = self.as_str();
-        s.rfind(":").and_then(|i| {
-            u16::from_str(&s[i+1..]).ok()
-        })
+    pub fn port_path(&self) -> Option<Port> {
```

Comment:
We have been using a _part suffix for this situation. We probably want to stick with that pattern.